rusoto/core/src/request.rs

```diff
@@ -172,19 +200,16 @@ impl fmt::Display for TlsError {
 
 /// Helper method for creating an http client with tls.
 /// Makes a `hyper` client with `NativeTls` for HTTPS support.
-pub fn default_tls_client() -> Result<Client, TlsError> {
-    let ssl = match NativeTlsClient::new() {
-        Ok(ssl) => ssl,
+pub fn default_tls_client(handle: &Handle) -> Result<Client<HttpsConnector<HttpConnector>>, TlsError> {
+    let connector = match HttpsConnector::new(4, handle) {
```

Comment:
Is this making a new HTTPS connection pool of size 4?

Reply:
This is actually the number of DNS Resolver threads: https://dabo.guru/rust/screeps-api/hyper_tls/struct.HttpsConnector.html

Reply:
Ah, thanks, good to know.  Ensuring we keep the connection pool feature is important.  Does `hyper` async (0.11 and later) do that for us?

Reply:
Not sure actually. That's a good catch. I don't see anything in changelog about it. Would be worth testing though.

Reply:
Hyper >=0.11 still maintains a connection pool, no regression here ;)